lib/puppet/provider/service/systemd.rb

```diff
@@ -44,7 +44,11 @@ def self.instances
   # in the provider's believed state of the service and the actual state.
   # @param action [String,Symbol] One of 'enable', 'disable', 'mask' or 'unmask'
   def systemctl_change_enable(action)
-    output = systemctl(action, @resource[:name])
+    if action == :enable || action == :disable
+      output = systemctl(action, '--', @resource[:name])
+    else
+      output = systemctl(action, @resource[:name])
+    end
```

Comment:
Seems like we'd have the same problem when executing `systemctl mask -h-asd`? Maybe we should always include `--` instead of just enable and disable?

Reply:
I can make this change. Do you squash merge pull requests or should I rebase and squash into my original commit?

Reply:
Best if you squash the commits and force push to your topic branch. Thanks!